python/lsst/daf/butler/registries/sqlPreFlight.py

```diff
@@ -0,0 +1,185 @@
+        return a set of data unit values.
+
+        Returned set consists of combinations of units participating in data
+        transformation from ``neededDatasetTypes`` to ``futureDatasetTypes``,
+        restricted by existing data and filter expression.
+
+        Parameters
+        ----------
+        collections : `list` of `str`
+            An ordered `list` of collections indicating the Collections to
+            search for Datasets.
+        expr : `str`
+            An expression that limits the `DataUnit`\ s and (indirectly) the
+            Datasets returned.
+        neededDatasetTypes : `list` of `DatasetType`
+            The `list` of `DatasetType`\ s whose instances should be included
+            in the graph and limit its extent.
+        futureDatasetTypes : `list` of `DatasetType`
+            The `list` of `DatasetType`\ s whose instances may be added to the
+            graph later, which requires that their `DataUnit` types must be
+            present in the graph.
+
+        Returns
+        -------
+        header : `tuple` of `tuple`
+            Length of tuple equals the number of columns in the returned
+            result set. Each item is a tuple with two elements - DataUnit
+            name (e.g. "Visit") and unit value name (e.g. "visit").
+        rows : sequence of `tuple`
+            Result set, this can be a single-pass iterator. Each tuple
+            contains unit values corresponding to units in a header.
+        """
+
+        # for now only a single collection is supported
+        if len(collections) != 1:
+            raise ValueError("Only single collection is supported by makeDataGraph()")
+        collection = collections[0]
+
+        # Collect unit names in both input and output dataset types
+        allUnits = set(itertools.chain.from_iterable(dsType.dataUnits for dsType in neededDatasetTypes))
+        allUnits.update(itertools.chain.from_iterable(dsType.dataUnits for dsType in futureDatasetTypes))
+        _LOG.debug("allUnits: %s", allUnits)
+
+        # All DataUnit instances in a subset that we need
+        allDataUnits = {unitName: self._schema.dataUnits[unitName] for unitName in allUnits}
+
+        # potentially we have to extend units set with the "required" superset
+        # from schema, for now just assume that set is full.
+
+        # Temporary workaround for AbstractFilter "view" which is not defined in schema yet,
+        # make a subquery for it and use it as an alias for that sub-query
+        dataUnit = allDataUnits.get("AbstractFilter")
+        if dataUnit and dataUnit.table is None:
+            # make a copy, do not change object in schema
+            dataUnit = copy.copy(dataUnit)
+            filterTable = self._schema.dataUnits["PhysicalFilter"].table
+            subquery = select([filterTable.c["abstract_filter"]]).distinct()
+            dataUnit._table = subquery.alias("AbstractFilter")
+            allDataUnits["AbstractFilter"] = dataUnit
```

Comment:
This workaround can be removed after you rebase on master; an `AbstractFilter` view was added on [DM-14428](https://jira.lsstcorp.org/browse/DM-14334).